boost-common/src/main/java/io/openliberty/boost/common/config/BoosterConfigurator.java

```diff
@@ -0,0 +1,107 @@
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.reflections.Reflections;
+
+import io.openliberty.boost.common.BoostException;
+import io.openliberty.boost.common.BoostLoggerI;
+import io.openliberty.boost.common.boosters.AbstractBoosterConfig;
+
+public class BoosterConfigurator {
+
+    /**
+     * take a list of pom boost dependency strings and map to liberty features
+     * for config. return a list of feature configuration objects for each found
+     * dependency.
+     * @param dependencies
+     * @param logger
+     * @return
+     * @throws BoostException
+     * @throws InstantiationException
+     * @throws IllegalAccessException
+     * @throws IllegalArgumentException
+     * @throws InvocationTargetException
+     * @throws NoSuchMethodException
+     * @throws SecurityException
+     */
+    public static List<AbstractBoosterConfig> getBoosterPackConfigurators(Map<String, String> dependencies,
+            BoostLoggerI logger) throws BoostException, InstantiationException, IllegalAccessException, IllegalArgumentException, InvocationTargetException, NoSuchMethodException, SecurityException {
+
+        List<AbstractBoosterConfig> boosterPackConfigList = new ArrayList<AbstractBoosterConfig>();
+        
+        Reflections reflections = new Reflections("io.openliberty.boost.common.boosters");
+
+        Set<Class<? extends AbstractBoosterConfig>> allClasses = reflections.getSubTypesOf(AbstractBoosterConfig.class);
+        for(Class<? extends AbstractBoosterConfig> boosterClass : allClasses) {
+            if (dependencies.containsKey(AbstractBoosterConfig.getCoordindates(boosterClass))) {
+                Constructor<?> cons = boosterClass.getConstructor(Map.class, BoostLoggerI.class);
+                Object o = cons.newInstance(dependencies, logger);
+                if(o instanceof AbstractBoosterConfig) {
+                    boosterPackConfigList.add((AbstractBoosterConfig)o);
+                }
+                else {
+                    throw new BoostException("Found a class in io.openliberty.boost.common.boosters that did not extend AbstractBoosterConfig. This should never happen.");
+                }
+            }
+        }
+        
+        return boosterPackConfigList;
+    }
+
+    public static void generateLibertyServerConfig(String libertyServerPath,
+            List<AbstractBoosterConfig> boosterPackConfigurators, String warName, BoostLoggerI logger) throws Exception {
+
+        LibertyServerConfigGenerator serverConfig = new LibertyServerConfigGenerator(libertyServerPath, logger);
+
+        // Loop through configuration objects and get features and XML config
+        // (if any)
+        for (AbstractBoosterConfig configurator : boosterPackConfigurators) {
+            serverConfig.addFeature(configurator.getFeature());
+            serverConfig.addBoosterConfig(configurator);
+        }
```

Comment:
One of the items I was working on for the mpConfig support came from a comment about collapsing these two separate calls into one action - as they both are adding to the server.xml. addFeature is a local method that adds the feature string retrieved by the configurator to the server.xml doc object as a feature stanza in the featureManager block. addBoosterConfig is a local method that takes a configurator object and in turn calls that configurator object with a reference to the same doc object to be updated with that configurator's config stanzas. Should both the configurator's feature name and its config stanzas (if there are any) be written out at the same time in a single call?